opm/test_util/summaryIntegrationTest.cpp

```diff
@@ -0,0 +1,417 @@
+	}
+  
+	if(spikePrev&&spikeCurrent){
+       
+	    std::cout << "For keyword " << keyword << " at time step " << (*referenceVec)[ivar] <<std::endl; 
+	    OPM_THROW(std::invalid_argument, "For keyword " << keyword << " at time step " << (*referenceVec)[ivar] << ", wwo deviations in a row exceed the limit. Not a spike value. Integration test fails." );
+	}
+	if(errorOccurrences > this->spikeLimit){
+	    std::cout << "For keyword " << keyword << std::endl; 
+	    OPM_THROW(std::invalid_argument, "For keyword " << keyword << " too many spikes in the vector. Integration test fails.");
+	}
+    }
+
+}
+
+
+WellProductionVolume IntegrationTest::getSpecificWellVolume(std::vector<double> timeVec1, std::vector<double> timeVec2, const char* keyword){
+    std::vector<double> dataVec1, dataVec2;
+    getDataVecs(dataVec1,dataVec2,keyword); 
+    chooseReference(timeVec1, timeVec2,dataVec1,dataVec2);
+      
+    return getWellProductionVolume(keyword);
+    
+   
+}
+
+bool IntegrationTest::checkUnits(const char * keyword){
+
+const smspec_node_type * node1 = ecl_sum_get_general_var_node (ecl_sum_fileShort ,keyword);
+const smspec_node_type * node2 = ecl_sum_get_general_var_node (ecl_sum_fileLong ,keyword);
+
+ if(strcmp(smspec_node_get_unit(node1),smspec_node_get_unit(node2)) == 0){
+     return true; 
+ }
+ return false; 
+}
+
+
+double IntegrationTest::integrate(std::vector<double> *timeVec, std::vector<double> *dataVec){
+    double totalSum = 0;
+    double width, height;
+    if(timeVec->size() != dataVec->size()){
+	OPM_THROW(std::runtime_error, "The size of the time vector does not match the size of the data vector.");
+    }
+    for(size_t i = 0; i < timeVec->size()-1; i++){
+	width = (*timeVec)[i+1] - (*timeVec)[i];
+	height = (*dataVec)[i+1];
+	totalSum += getRectangleArea(height, width);
+    }
+    return totalSum; 
+}
+
+double IntegrationTest::integrateError(std::vector<double> *timeVec1, std::vector<double> *dataVec1, std::vector<double> *timeVec2, std::vector<double> *dataVec2){
+
+//When the data corresponds to a rate the integration will become a Riemann sum. This function calculates the Riemann sum of the error. The reason why a Riemann sum is used is because of the way the data is written to file. When a change occur(e.g. change of a rate), the data (value and time) is written to file, THEN the change happenes in the simulator, i.e. we will notice the change at the next step. Keep in mind that the summary vector is NOT a continous curve, only points of data (time, value). We have to guess what happenes between the data points, we do this by saying: "There are no change, the only change happenes at the data points.". As stated above, the value of this constant "heigth" of the rectangle corresponds to the value of the last time step. Thus we have to use the "right hand side value of the rectangle as height (*someDataVector)[ivar] and NOT (*someDataVector)[ivar-1] (which intuition is saying is the correct value to use). 
```

Comment:
This must be a new OPM record for long lines :D  Almost 1k characters.

Please break the lines, and remember, you are allowed to use paragraphs.  Example:

```c++
// When the data corresponds to a rate the integration will become a Riemann
// sum.  This function calculates the Riemann sum of the error.  The reason why
// a Riemann sum is used is because of the way the data is written to file.
// When a change occur (e.g. change of a rate), the data (value and time) is
// written to file, THEN the change happens in the simulator, i.e., we will
// notice the change at the next step.
//
// Keep in mind that the summary vector is NOT a continuous curve, only points
// of data (time, value).  We have to guess what happens between the data
// points, we do this by saying: "There are no change, the only change happens
// at the data points."  As stated above, the value of this constant "height" of
// the rectangle corresponds to the value of the last time step.  Thus we have
// to use the "right hand side value of the rectangle as height
//
// (*someDataVector)[ivar] instead of (*someDataVector)[ivar-1]
//
// (which intuition is saying is the correct value to use).
```